#!/bin/sh /usr/share/dpatch/dpatch-run
## 08-SecBull-TYPO3-SA-2009-001-10134.dpatch by Christian Welzel <gawain@camlann.de>
##
## DP: fix for TYPO3 Security Bulletin TYPO3-SA-2009-001: Multiple vulnerabilities in TYPO3 Core

@DPATCH@

diff -Naur typo3_src-4.0.9/typo3/sysext/indexed_search/mod/index.php typo3_src-4.0.10/typo3/sysext/indexed_search/mod/index.php
--- typo3_src-4.0.9/typo3/sysext/indexed_search/mod/index.php 2007-12-10 21:44:25.000000000 +0100
+++ typo3_src-4.0.10/typo3/sysext/indexed_search/mod/index.php 2008-06-11 10:43:07.000000000 +0200
@@ -300,7 +300,7 @@
 			$grListRec = $this->getGrlistRecord($row["phash"]);
 			$recList[] = array(
 				$row["data_page_id"].($row["data_page_type"]?"/".$row["data_page_type"]:""),
-				t3lib_div::fixed_lgd($row["item_title"],30),
+				htmlspecialchars(t3lib_div::fixed_lgd($row["item_title"],30)),
 				t3lib_div::formatSize($row["item_size"]),
 				$this->getNumberOfWords($row["phash"]),
 				t3lib_BEfunc::datetime($row["item_mtime"]),
@@ -367,7 +367,7 @@
 			$cHash = count(unserialize($row["cHashParams"])) ? $this->formatCHash(unserialize($row["cHashParams"])) : "";
 			$grListRec = $this->getGrlistRecord($row["phash"]);
 			$recList[]=array(
-				t3lib_div::fixed_lgd($row["item_title"],30),
+				htmlspecialchars(t3lib_div::fixed_lgd($row["item_title"],30)),
 				t3lib_div::formatSize($row["item_size"]),
 				$this->getNumberOfWords($row["phash"]),
 				t3lib_BEfunc::datetime($row["item_mtime"]),
@@ -378,7 +378,7 @@
 				$row["pcount"],
 				$cHash,
 				$row["phash"],
-				t3lib_div::fixed_lgd($row["data_filename"],100)
+				htmlspecialchars(t3lib_div::fixed_lgd($row["data_filename"],100))
 			);
 
 			if ($row["pcount"]>1)	{
